### DISCUSSION TITLE
Secure user Metadata

### DISCUSSION THREAD

**Discussion by MadL1me** [zitadel/zitadel#7709] [General]
Hi, I have a question targeting feature of "user metadata". Is there a way to create a metadata, which cannot be accessed using user access_token, but can be accessed using admin token or application (for example, and API)? 

Use case: I want to store secrets for user (for example Webhook Secret), but user should not be able to access this secrets himself. Only authorized client or admin should be able to do so

**Comment by dakshitha** [comment]
Hey! Unfortunately, this is currently not supported. The user can always get his own metadata with his access token. 
It will be possible with User Schemas, where they will be able to create their own schema and define properties to be only readable by "owners" and not the user itself.

User Schema Documentation -  https://zitadel.com/docs/guides/manage/customize/user-schema 
Current state of implementation of User Schema - https://github.com/zitadel/zitadel/issues/6433
